### PR TITLE
wg-installer: add wg-installer

### PR DESCRIPTION
**Pull request by PolynomialDivision**
This tool can be used to automatically create wireguard tunnels. Using rpcd a new wireguard interface is created on the server where the client can connect to.

Wiregurad server automatically installs a user and associated ACL to use the wireguard-installer-server features. The user is called wginstaller and so is the password.

Get Usage:
```
  wg-client-installer get_usage --ip 127.0.0.1 --user wginstaller
	--password wginstaller
```

Register Interface:
```
  wg-client-installer register --ip 127.0.0.1 --user wginstaller
         --password wginstaller --bandwidth 10 --mtu 1400
```

Maintainer: me
Compile tested: trunk
Run tested: trunk

### CODE REVIEW

**Comment by PolynomialDivision** [comment]
This installer is in early progress stage.

File: net/wg-installer/wg-server/lib/wg_functions.sh
Comment (doverride) [inline net/wg-installer/wg-server/lib/wg_functions.sh:46]: should babel not be a dependency then?

Comment (PolynomialDivision) [reply net/wg-installer/wg-server/lib/wg_functions.sh:46]: Whups, forgot about that we reload babeld at this point. :S I will remove it and make it independent of babeld.

Comment (PolynomialDivision) [reply net/wg-installer/wg-server/lib/wg_functions.sh:46]: we use this for creating mesh links via wireguard with babeld.

Comment (PolynomialDivision) [reply net/wg-installer/wg-server/lib/wg_functions.sh]: I will find another way to reload babeld

File: net/wg-installer/wg-client/lib/rpcd_ubus.sh
Comment (neheb) [inline net/wg-installer/wg-client/lib/rpcd_ubus.sh:3]: function? this is not bash.

Comment (PolynomialDivision) [reply net/wg-installer/wg-client/lib/rpcd_ubus.sh:3]: You can do both? If u like I can switch to the c-style notation.

Comment (neheb) [reply net/wg-installer/wg-client/lib/rpcd_ubus.sh:3]: I'm pretty sure shellcheck complains on this. function is a bashism IIRC.

Comment (PolynomialDivision) [reply net/wg-installer/wg-client/lib/rpcd_ubus.sh]: I changed that. Give me some time to run-test again. ;) No access to OpenWrt router for 1 hour.

**Review by neheb — APPROVED** [review]

**Comment by PolynomialDivision** [comment]
@doverride I removed the babeld dependency. Could we go on merging this? Your approve is missing. ;)

**Comment by doverride** [comment]
My review does not count towards merge-ability (no write access), sorry

**Comment by PolynomialDivision** [comment]
Since, I have now commit access, I'm allowed to merge this (since it has 1 approve)? :)

Do u merge that locally, or do u use the github button to merge a PR? I looked again at the conribution guidelines, and nothing is described there. :/

**Comment by champtar** [comment]
Some people prefer rebase but yes with 1 approval you can go ahead. There is less chance of pushing unrelated commit using github rebase button, and not sure what happens to the PR if you push the commit to master (maybe Github is clever enough, maybe not)

**Comment by PolynomialDivision** [comment]
> Some people prefer rebase but yes with 1 approval you can go ahead. There is less chance of pushing unrelated commit using github rebase button, and not sure what happens to the PR if you push the commit to master (maybe Github is clever enough, maybe not)

I always see merge commits (so just here not in the openwrt.git)? I prefer rebase to, but I wanted to merge-commit now, since I thought is would be the preferred style?

**Comment by neheb** [comment]
There are lots of opinions on this. I did rebase merges at one point until I got complaints that the commits I merged were no longer GPG signed. Technically merging provides the "real" history but I don't know how valuable that is.

**Comment by PolynomialDivision** [comment]
So I can just do what I personally prefer? Then I will just do a rebase merge. ;)

**Comment by neheb** [comment]
Sure

**Comment by PolynomialDivision** [comment]
Here is a hotplug.d example, to reload interfaces:

```
#!/bin/sh

# check if wireguard
if [ "${DEVTYPE}" != "wireguard" ]; then
	exit 0
fi

# check if correct naming
slicedint=$(echo $INTERFACE | cut -c1-3)
if [ "${slicedint}" != "wg_" ]; then
	exit 0
fi

if [ "${ACTION}" == "add" ]; then
	uci add babeld interface
	uci set babeld.@interface[-1].ifname="${INTERFACE}"
	uci commit
	/etc/init.d/babeld reload
fi

if [ "${ACTION}" == "remove" ]; then
	i=0
	while uci get babeld.@interface[$i] &> /dev/null ; do
		if [ "$(uci get babeld.@interface[$i].ifname)" == "${INTERFACE}" ]; then
			uci delete babeld.@interface[$i]
		fi
		i=$((i+1));
	done
	uci commit
	/etc/init.d/babeld reload
fi
```

**Pull request merged**
